src/main/java/org/fungover/mmotodo/team/TeamService.java

```diff
@@ -0,0 +1,116 @@
+        Team team = teamRepository.findById(teamId).orElseThrow(() -> new TeamNotFoundException("Team not found"));
+        teamRepository.delete(team);
+        return "Team with id:" + teamId + " deleted";
+    }
+
+    @Transactional
+    public Team createTeam(@Valid TeamCreateDto teamCreate) {
+        // Check if the team with the same name already exists
+        if (teamRepository.existsByName(teamCreate.name())) {
+            throw new RuntimeException("Team already exists");
+        }
+
+        Team team = new Team();
+        team.setName(teamCreate.name());
+        return teamRepository.save(team);
+
+    }
+
+    @Transactional
+    public Team updateTeam(TeamUpdateDto teamUpdate) {
+
+        Team team = teamRepository.findById(teamUpdate.id()).orElseThrow(() -> new TeamNotFoundException("Team not found"));
+        team.setName(teamUpdate.name());
+        return teamRepository.save(team);
+    }
+
+    @Transactional
+    public String addUserToTeam(int teamId, int userId) {
+        Team team = teamRepository.findById(teamId).orElseThrow(() -> new TeamNotFoundException("Team not found"));
+        User user = userRepository.findById(userId).orElseThrow(() -> new TeamNotFoundException("Team not found"));
+        if (team.getUsers().contains(user)) {
+            throw new RuntimeException("User already present in team");
+        }
+        team.addUser(user);
+        teamRepository.save(team);
+        return"user added";
+    }
+
+    @Transactional
+    public String removeUserFromTeam(int teamId, int userId) {
+        Team team = teamRepository.findById(teamId).orElseThrow(() -> new TeamNotFoundException("Team  not found"));
+        User user = userRepository.findById(userId).orElseThrow(() -> new EntityNotFoundException("User with ID " + userId + " not found"));
+        if (!team.getUsers().contains(user)) {
+            throw new RuntimeException("User not present in team: " + team.getName());
+        }
+        team.removeUser(user);
+        teamRepository.save(team);
+        return "user removed";
+    }
+
+    @Transactional
+    public String addTaskToTeam(int teamId, int taskId) {
+        Team team = teamRepository.findById(teamId).orElseThrow(() -> new TeamNotFoundException("Team not found"));
+        Task task = taskRepository.findById(taskId).orElseThrow(() -> new EntityNotFoundException("Task not found"));
+        if (team.getTasks().contains(task)) {
+            throw new RuntimeException("This task has  already been assigned to this team");
+        }
+        team.addTask(task);
+        teamRepository.save(team);
+        return "task added";
+
+    }
+
+    @Transactional
+    public String removeTaskFromTeam(int teamId, int taskId) {
+        Team team = teamRepository.findById(teamId).orElseThrow(() -> new TeamNotFoundException("Team  not found"));
+        Task task = taskRepository.findById(taskId).orElseThrow(() -> new EntityNotFoundException("Task not found"));
```

Comment:
A lot of exception code that's repeating. You could move the message to the TeamNotFoundException class instead of having to write out the message everytime you write a throwable. Could also be nice to include what team/task that's not being found. Otherwise good exceptions.

```
public class TeamNotFoundException extends ResourceNotFoundException{
    public TeamNotFoundException(int teamId) {
        super("Team not found with ID: " + teamId);
    }
}
```